packages/react/src/Pillar/Pillar.tsx

```diff
@@ -0,0 +1,119 @@
+import React, {forwardRef} from 'react'
+import {isFragment} from 'react-is'
+import clsx from 'clsx'
+import {Heading, HeadingTags, HeadingProps, Text} from '..'
+import type {BaseProps} from '../component-helpers'
+import {Colors} from '../constants'
+
+/**
+ * Design tokens
+ */
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/pillar/base.css'
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/pillar/colors-with-modes.css'
+
+/**
+ * Main stylesheet (as a CSS Module)
+ */
+import styles from './Pillar.module.css'
+import {Icon as IconProps} from '@primer/octicons-react'
+
+export const PillarIconColors = Colors
+
+export const defaultPillarIconColor = PillarIconColors[0]
+export type PillarProps = {
+  /**
+   * Valid children include Pillar.Image, Pillar.Heading, and Pillar.Description
+   */
+  children:
+    | React.ReactNode
+    | React.ReactElement<PillarIconProps>
+    | React.ReactElement<PillarHeadingProps>
+    | React.ReactElement<PillarDescriptionProps>
+
+  /**
+   * Aligns the pillar content
+   */
+  align?: 'start' | 'center'
+} & BaseProps<HTMLDivElement> &
+  React.ComponentPropsWithoutRef<'div'>
```

Comment:
Should we maybe support an isomorphic `as` prop so pillars can be set to a more semantic element such as `article`?